### DISCUSSION TITLE
New Open Source License - Thoughts?

### DISCUSSION THREAD

**Discussion by wmthornton-dev** [community/community#147254] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

General

### Body

I've recently (well, 2021) written a new license for open source software projects I like to call the Public Benefit Zero Copyright License (PBZC). The idea of this license is similar to the CC0, but in many ways more succinctly dedicates open source software works to the public domain, for the public benefit. It is meant to be "cancerous", much in the way the GPL is i.e. the license cannot be revoked or changed down the road by another developer. I would love input from the community on the license itself and genuine feedback (I don't want arguments about the GPL being the best license, etc.).  The license itself is located on both of my Github profiles, but most recently updated here: https://github.com/wmthornton-dev/PBZC.git

I am also including the text of the license below:

PUBLIC BENEFIT ZERO COPYRIGHT LICENSE v. 1.0
SPDX Identifier: <TBD>

Revision Date: October 18th, 2021

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is expressly prohibited.

Definitions:

1) "This License" refers to version 1.0 of the Public Benefit
   Zero Copyright License.

2) "Copyright" means any laws intended to protect software and associated
   works, such as written texts and documentation, but this definition is not 
   limited and shall be construed to include other works as defined by local 
   or international laws.

3) "This Software" or "this software" refers to any ostensibly
   copyrightable work to which This License has been assigned.

4) "Derivative Software" or "derivative software" refers to any
   ostensibly copyrightable work within which any portion was derived
   from This Software or any work subject to This License.

5) "General Public Benefit" or "general public benefit" shall mean Free
   Open Source Software and/or Commercial Open Source Software, including
   operating systems and applications software which fosters technological 
   and/or educational advancement. 

6) "Public Domain" or "public domain" shall mean This Software or
   Derivative Software is entirely free from ownership such as copyright,
   trademark, or patent.

7) "Free Open Source Software" shall mean This Software or Derivative
   Software that licensed to be free to use, modify, and distribute.

8) "Commercial Open Source Software" shall mean This Software or Derivative
   Software which is designed to interact with closed, proprietary software
   in such a way that not all code is publicly available. Software released under 
   this definition may be sold as part of a larger commercial product
   but a public repository must be maintained which includes This Software or
   Derivative Software, in addition to dedicated documentation regarding all API 
   endpoints and system calls such that a Third-Party developer can understand what 
   This Software or Derivative Software is doing when interacting with 
   proprietary source code, binaries, libraries and APIs. Additionally, This
   Software or Derivative Software must be made free to use, modify or
   distribute in derivative works by Third-Party developers.

This is free and unencumbered software released into the public domain,
and intended to be kept in the public domain in perpetuity.

Anyone is free to copy, modify, publish, use, compile, sell, or
distribute this software, either in source code form or as a compiled
binary, for any purpose, commercial or non-commercial, and by any
means provided:

1) This Software or Derivative Software is intended to inure to the
   General Public Benefit, and

2) They do not assign additional licensing or attribution
   requirements to This Software or Derivative Software, and

3) This Software or Derivative Software remains subject to this
   revision of the Public Benefit Zero Copyright License or its'
   superseding revision, whichever is most recent at the time of
   development, and

4) This Software or Derivative Software is publicly released as Free
   Open Source Software and/or as Commercial Open Source Software, as
   appropriate, and

5) A copy of This License is included with all distributions of This
   Software or Derivative Software.

In jurisdictions that recognize copyright laws, the author or authors
of this software dedicate any and all copyright interest in This
Software to the public domain subject to the provisions above.
We make this dedication for the benefit of the public at large and to 
the detriment of our heirs and successors. We intend this dedication to
be an overt act of relinquishment in perpetuity of all present and future
rights to this software under copyright law.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
OTHER DEALINGS IN THE SOFTWARE.

For more information, please refer to <https://github.com/wmthornton/PBZC.git>

EDIT: 

I updated the license text this morning as a result of some thoughts regarding clarity and potential issues and holes in the license. Here is the updated text:

PUBLIC BENEFIT ZERO COPYRIGHT LICENSE v. 2.0
SPDX Identifier: <TBD>

Revision Date: December 17th, 2024

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is expressly prohibited.

Definitions:

1) "This License" refers to version 1.0 or later of the Public Benefit
Zero Copyright License, whichever is the current release at the time of development.

2) "Copyright" means any laws intended to protect software and associated
works, such as written texts and documentation, however this definition is 
not limited to, and shall be construed to include, other related works as 
defined by local or international laws at the time of development.

3) "This Software" or "this software" refers to any ostensibly
copyrightable work to which This License has been assigned, including 
written texts and documentation, source code, computer binaries, and all 
other related materials and/or distributions.

4) "Derivative Software" or "derivative software" refers to any
ostensibly copyrightable work within which any portion was derived
from This Software or any work subject to This License.

5) "General Public Benefit" or "general public benefit" shall mean any Free
Open Source Software and/or Commercial Open Source Software, including operating 
systems and applications software from which the general public, developers, governments, 
individuals and any other such groups can gain use, knowledge, experience or any other 
benefit from, the definition of which is at their sole discretion.

6) "Public Domain" or "public domain" shall mean This Software or
Derivative Software is entirely free from ownership such as copyright, trademark, 
or patent.

7) "Free Open Source Software" shall mean This Software or Derivative
Software that licensed to be free to use, modify, and distribute.

8) "Commercial Open Source Software" shall mean This Software or Derivative
Software which is designed to interact with closed, proprietary software in such a way 
that not all code is publicly available. Software released under this definition may be sold 
as part of a larger commercial product, however a public repository must be maintained which 
includes This Software or Derivative Software, in addition to dedicated documentation regarding 
all API endpoints and system calls such that a Third-Party developer can understand what This Software 
or Derivative Software is doing when interacting with proprietary source code, binaries, libraries 
and APIs. Additionally, This Software or Derivative Software must be made free to use, modify or
distribute in derivative works by Third-Party developers.

This is free and unencumbered software released into the public domain,
and intended to be kept in the public domain in perpetuity.

Anyone is free to copy, modify, publish, use, compile, sell, or
distribute this software, either in source code form or as a compiled
binary, for any purpose, commercial or non-commercial, and by any
means provided:

1) This Software or Derivative Software is intended to inure to the
   General Public Benefit, and

2) They do not assign additional licensing or attribution
   requirements to This Software or Derivative Software, and

3) This Software or Derivative Software remains subject to this
   revision of the Public Benefit Zero Copyright License or its'
   superseding revision, whichever is most recent at the time of
   development, and

4) This Software or Derivative Software is publicly released as Free
   Open Source Software and/or as Commercial Open Source Software, as
   appropriate, and

5) A copy of This License is included with all distributions of This
   Software or Derivative Software.

In jurisdictions that recognize copyright laws, the author or authors
of this software dedicate any and all copyright interest in This
Software to the public domain subject to the provisions above.
We make this dedication for the benefit of the public at large and to 
the detriment of our heirs and successors. We intend this dedication to
be an overt act of relinquishment in perpetuity of all present and future
rights to this software under copyright law.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
OTHER DEALINGS IN THE SOFTWARE.

For more information, please refer to <https://github.com/wmthornton-dev/PBZC.git>

**Comment by Logophoman** [comment]
Hmm it looks very nice, but don't you think it is quite similar to AGPL3 that basically says it's open source and you must publish every change but are free to use it commercially? 🤔😊

**Reply by wmthornton-dev** [reply]
Thank you for your response.

No, I don't think it's very similar to the AGPL3 in that the AGPL3 still asserts copyright mechanisms but conveys use rights under those mechanisms to others. By contrast, PBZC specifically renounces all copyright claims for both the developer and all heirs, assigns, etc downstream in perpetuity, which means that no one can ever claim copyright over the source code or derivatives. The idea is that the license fosters technological advancement and innovation without copyright issues getting in the way which is vital for education.

**Reply by Logophoman** [reply, marked as answer]
Nice, didn't know that yet and I love the concept of waving 100% of all claims. Have you tried posting it on Reddit, Daily.dev or considerd making a YouTube video to raise awareness? I feel like this needs more reach...

**Reply by wmthornton-dev** [reply]
I haven't as of yet. I submitted it to OSI for approval as an officially recognized license, but that can take a few months to complete the process.  
Part of the problem I face is that I'm a lone person, and the creator of the license, trying to get the word out. As you know, the OS community is very opinionated by nature, and adding any license that isn't one of the major ones (GPL, MIT, Apache) but seeks to address issues with those licenses or take a different approach, faces a lot of backlash.
